src/Plugin/Validation/Constraint/DecoupledAuthUserNameConstraintValidator.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\user\Plugin\Validation\Constraint\UserNameConstraintValidator.
+ */
+
+namespace Drupal\decoupled_auth\Plugin\Validation\Constraint;
+
+use Drupal\user\Plugin\Validation\Constraint\UserNameConstraintValidator;
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * Validates the DecoupledAuthUserName constraint.
+ */
+class DecoupledAuthUserNameConstraintValidator extends UserNameConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($items, Constraint $constraint) {
+    if (!isset($items) || !$items->value) {
```

Comment:
Shouldn't this only return here if the password is also empty? What is to stop someone from just unsetting the username? Would this result in someone logging in with a blank username and just a password? Or could this lead to a user accidentally blocking themselves from the site?

We also want to make sure the the register form doesn't let unauthenticated users from adding endless userless-users.